hs_rest_api2/metadata.py

```diff
@@ -0,0 +1,238 @@
+import json
+import os
+from datetime import datetime, timezone
+
+from django.db import transaction
+from django.http import JsonResponse, HttpResponse
+from drf_yasg.utils import swagger_auto_schema
+from pydantic import ValidationError as PydanticValidationError
+from rest_framework.exceptions import ValidationError
+
+from hs_core.hydroshare.hs_bagit import create_bag_metadata_files
+from rest_framework.decorators import api_view
+
+from hs_core.models import ResourceFile
+from hs_core.views import ACTION_TO_AUTHORIZE
+from hs_core.views.utils import authorize
+from hs_file_types.utils import ingest_logical_file_metadata
+from hs_rest_api2 import serializers
+from hs_rest_api2.serializers import ResourceMetadataInForbidExtra
+
+from hsmodels.schemas import ResourceMetadata
+
+
+def load_metadata(istorage, file_with_path):
+    from hsmodels.schemas import load_rdf
+    with istorage.open(file_with_path) as f:
+        metadata = load_rdf(f.read())
+        return metadata
+
+
+def resource_metadata(resource):
+    file_with_path = resource.scimeta_path
+    istorage = resource.get_irods_storage()
+    return load_metadata(istorage, file_with_path)
+
+
+def ingest_resource_metadata(resource, incoming_metadata):
+    from hsmodels.schemas import rdf_graph
+    r_md = resource_metadata(resource).dict()
+    try:
+        incoming_r_md = ResourceMetadataInForbidExtra(**incoming_metadata)
+    except PydanticValidationError as e:
+        raise ValidationError(e)
+    # merge existing metadata with incoming, incoming overrides existing
+    merged_metadata = {**r_md,
+                       **incoming_r_md.dict(exclude_defaults=True),
+                       "modified": datetime.now(timezone.utc).isoformat()}
+    res_metadata = ResourceMetadata(**merged_metadata)
+
+    graph = rdf_graph(res_metadata)
+    try:
+        with transaction.atomic():
+            resource.metadata.delete_all_elements()
+            resource.metadata.ingest_metadata(graph)
+    except:
+        # logger.exception("Error processing resource metadata file")
+        raise
+    create_bag_metadata_files(resource)
```

Comment:
This shouldn't be happening here, we really just need to create the XML files for the updated metadata.

Reply:
updated to save just the xml file that was changed